frameworks/swift/xctest/PrintObserver.swift

```diff
@@ -0,0 +1,80 @@
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//
+//  PrintObserver.swift
+//  Prints test progress to stdout.
+//
+
+#if os(Linux) || os(FreeBSD)
+    import Foundation
+#else
+    import SwiftFoundation
+#endif
+
+/// Prints textual representations of each XCTestObservation event to stdout.
+/// Mirrors the Apple XCTest output exactly.
+internal class PrintObserver: XCTestObservation {
+    func testBundleWillStart(_ testBundle: Bundle) {}
+
+    func testSuiteWillStart(_ testSuite: XCTestSuite) {
+        if testSuite.name != "All tests" && testSuite.name != "bin.xctest" {
+          printAndFlush("<DESCRIBE::>\(testSuite.name)")
+        }
+    }
+
+    func getMethodName(_ testCase: XCTestCase) -> String {
+      if testCase.name.contains(".") {
+        return testCase.name.components(separatedBy: ".")[1]
+      }
+      return testCase.name
+    }
+
+    func testCaseWillStart(_ testCase: XCTestCase) {
+        printAndFlush("<IT::>\(getMethodName(testCase))")
+    }
+
+    func testCase(_ testCase: XCTestCase, didFailWithDescription description: String, inFile filePath: String?, atLine lineNumber: UInt) {
+        printAndFlush("<ERROR::>\(getMethodName(testCase)) : \(description)")
```

Comment:
It needs to replace every newline—I'm very new to Swift, so I don't know if that's how `terminator` works or not.

What you want to make sure is something like this:
```
<FAILED::>error on line 1 of foo
on line 2 of bar
```

Is rendered like this:

```
<FAILED::>error on line 1 of foo<:LF:>on line 2 of bar
```

This helps the runner to keep related blocks of text together.